src/data/feed/quandl.js

```diff
@@ -0,0 +1,146 @@
+import d3 from 'd3';
+
+//  https://www.quandl.com/docs/api#datasets
+export default function() {
+
+    var _columnNameMap = function(colName) {
+        colName = colName.replace(/[\s()-.]/g, '');
+        return colName.substr(0, 1).toLowerCase() + colName.substr(1);
+    };
+
+    var database = 'YAHOO',
+        dataset = 'GOOG',
+        apiKey = null,
+        start = null,
+        end = null,
+        rows = null,
+        descending = false,
+        collapse = null,
+        columnNameMap = _columnNameMap;
+
+    var quandl = function(cb) {
+        var params = [];
+        if (apiKey != null) {
+            params.push('api_key=' + apiKey);
+        }
+        if (start != null) {
+            params.push('start_date=' + start.toISOString().substring(0, 10));
+        }
+        if (end != null) {
+            params.push('end_date=' + end.toISOString().substring(0, 10));
+        }
+        if (rows != null) {
+            params.push('rows=' + rows);
+        }
+        if (!descending) {
+            params.push('order=asc');
+        }
+        if (collapse != null) {
+            params.push('collapse=' + collapse);
+        }
+
+        var url = 'https://www.quandl.com/api/v3/datasets';
+        url += ('/' + database + '/' + dataset + '/').replace(/\/\//g, '');
```

Comment:
I'm not sure it'll have the expected behaviour if the database has a
trailing slash. I'd just leave it up to the consumer to use the right values

On Fri, 9 Oct 2015 08:42 James Dunkerley <notifications@github.com> wrote:

> In src/data/feed/quandl.js
> <https://github.com/ScottLogic/d3fc/pull/595#discussion_r41604511>:
>
> > +        }
> > +        if (end != null) {
> > +            params.push('end_date=' + end.toISOString().substring(0, 10));
> > +        }
> > +        if (rows != null) {
> > +            params.push('rows=' + rows);
> > +        }
> > +        if (!descending) {
> > +            params.push('order=asc');
> > +        }
> > +        if (collapse != null) {
> > +            params.push('collapse=' + collapse);
> > +        }
> > +
> > +        var url = 'https://www.quandl.com/api/v3/datasets';
> > +        url += ('/' + database + '/' + dataset + '/').replace(/\/\//g, '');
>
> No real need for it, meant to trim off leading and trailing '/' in the
> database or dataset.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ScottLogic/d3fc/pull/595/files#r41604511>.
>